core/src/main/java/com/sequenceiq/cloudbreak/service/stack/DependentRolesHealthCheckService.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.InstanceStatus;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+
+/**
+ * Returns the list of Componenets on which the given component depends.
+ * roleDependencyMap is not an exhaustive list of dependencies.
+ * In any case where dependency is not defined, it will return  "UNDEFINED_DEPENDENCY"
+ */
+
+@Service
+public class DependentRolesHealthCheckService {
+
+    public static final String UNDEFINED_DEPENDENCY = "UNDEFINED_DEPENDENCY";
+
+    private static final Map<String, List<String>> ROLE_DEPENDENCIES = Map.ofEntries(
+            Map.entry("NODEMANAGER", List.of("RESOURCEMANAGER")));
+
+    /**
+     * Returns a list of Components on which the Source Component Depends
+     * or "UNDEFINED DEPENDENCY" if the dependency is not defined in the ROLE_DEPENDENCIES
+     * @param srcComponent takes the source component as input for which we need to find the dependency.
+     * @return List of components on which the input parameter depends or "UNDEFINED_DEPENDENCY" if dependency not defined
+     */
+
+    public List<String> getRoleDependency(String srcComponent) {
+        return ROLE_DEPENDENCIES.getOrDefault(srcComponent, Collections.singletonList(UNDEFINED_DEPENDENCY));
+    }
```

Comment:
Looks like this method is only used in unit tests, can be removed if so. JavaDoc can be added to `getUnhealthyDependentHostGroups` below

Reply:
I have still kept it because in future if the ROLE_DEPENDENCIES map is required than this can be used.
I will add javadoc for getUnhealthyDependentHostGroups